src/app/heroes/heroes.service.ts

```diff
@@ -1,17 +1,34 @@
 import { Injectable } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 
 import { Hero } from '../models/hero';
 
 @Injectable()
 export class HeroesService {
+  private params: HttpParams = new HttpParams();
 
   constructor(private http: HttpClient) { }
 
-  getHeroes(searchValue = '', pageNumber = 1): Observable<{count: number, heroes: Hero[]}> {
-    return this.http.get(`https://swapi.co/api/people/?page=${pageNumber}&search=${searchValue}`)
+  getHeroes(searchTerm, pageNumber): Observable<{count: number, heroes: Hero[]}> {
+    if (searchTerm) {
+      this.params = this.params.set('search', searchTerm);
+    }
+    
+    this.params = this.params.set('page', pageNumber);
+    
+    return this.http.get(`https://swapi.co/api/people/`, { params: this.params } )
+      .map((data: any) => ({
+        count: data.count,
+        heroes: data.results
+      }));
+  }
+
+  search(searchTerm) {
+    this.params = this.params.set('search', searchTerm);
+
+    return this.http.get(`https://swapi.co/api/people/`, { params: this.params })
```

Comment:
Don't add `/` at the end of the url, your url will look like `https://swapi.co/api/people/?page=1` and it's a little bit strange.